### PR TITLE
docs(ob_error): research and scheme documents of ob_error tool

### LINKED ISSUE #176: ob_error tool
Both Oracle and MySQL have their own error tools, named **oerr** and **perror** respectively. 

The error tool can return the information, reason and solution corresponding to the error code according to the error code entered by the user. It saves the trouble of looking up documents.

such as:
```
$ oerr ora 600

ora-00600: internal error code, arguments: [%s], [%s], [%s], [%s], [%s], [%s], [%s], [%s]

 *Cause:  This is the generic internal error number for Oracle program

​     exceptions.  This indicates that a process has encountered an

​     exceptional condition.

 *Action: Report as a bug - the first argument is the internal error number
```
```
$ perror 1231

MySQL error code MY-001231 (ER_WRONG_VALUE_FOR_VAR): Variable '%-.64s'

can't be set to the value of '%-.200s'
```
But OceanBase doesn't have such a tool at present.

For ease of use, such a tool is needed, which named **ob_error**.

### PR DESCRIPTION
**Pull request by a1iive**
This is the research and scheme documents of ob_error tool.
fixes #176

### CODE REVIEW

File: doc/ob_error-research-and-scheme-cn.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+> ERROR 1007 (HY000) : Can't create database '%s'; database exists
+
+ob的内部错误码为5158，而Oracle本身没有该错误类型，因此在Oracle模式下的错误输出为：
+
+> ORA-00600：internal error code, arguments: -5158, Can't create database '%.*s'; database exists
+
+
+
+MySQL模式在[1,4000)范围内有多个特殊的错误码，详见[文档](https://www.oceanbase.com/docs/oceanbase-database/oceanbase-database/V3.1.2/0001-3999#title-9k7-sdh-gch)，可能存在错误码相同，但含义不同的极少数特殊情况。
```
Comment (simonjoylet) [inline doc/ob_error-research-and-scheme-cn.md:184]: the discussion of MySQL mode better move to MySQL section.

File: doc/ob_error-research-and-scheme-cn.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+>
+> Solution: to be continue
+
+# 方案实现
+
+1.在ob_errno.def内增加C2/C3字段分别表示错误原因和错误解决方法
+
+2.具体错误原因和解决方法暂时用通用字段“to be continue”代替，后续由研发同学更新，更新方法可参考ob_errno.def文件的注解，更新后需重新执行gen_ob_errno.pl
+
+1.增加Oracle模式错误码到ob错误码，以及mysql模式错误码（小于4000）到ob错误码的映射关系
```
Comment (simonjoylet) [inline doc/ob_error-research-and-scheme-cn.md:365]: wrong sequence here?

File: doc/ob_error-research-and-scheme-cn.md [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+
+2.举例：
+
+如果得到`ORA-7300`的错误码，则“ora”是设施，而“7300”是错误。所以输入***oerr ora 7300***。
+
+如果得到`LCD-111`的错误码，则输入***oerr lcd 111***。
+
+1.输出：
+
+> $ oerr [ora 600](https://www.orafaq.com/wiki/ORA-00600)
```
Comment (Amber1990Zhang) [inline doc/ob_error-research-and-scheme-cn.md:19]: delete the extra content in ()

**Review by Amber1990Zhang — APPROVED** [review]

**Review by LINxiansheng — APPROVED** [review]
LGTM

**Comment by LINxiansheng** [comment]
I merged it and changed the fold path
6b502f35f4b7a75fe40ec3cdb884f5d98cf0c2e0

**Comment by LINxiansheng** [comment]
Thanks for your contribution

**Pull request closed without merging**